packages/block-library/src/navigation/edit/index.js

```diff
@@ -561,33 +566,43 @@ function Navigation( {
 								</div>
 							</>
 						) }
-						<h3>{ __( 'Overlay Menu' ) }</h3>
-						<ToggleGroupControl
-							__nextHasNoMarginBottom
-							label={ __( 'Configure overlay menu' ) }
-							value={ overlayMenu }
-							help={ __(
-								'Collapses the navigation options in a menu icon opening an overlay.'
-							) }
-							onChange={ ( value ) =>
-								setAttributes( { overlayMenu: value } )
-							}
-							isBlock
-							hideLabelFromVision
-						>
-							<ToggleGroupControlOption
-								value="never"
-								label={ __( 'Off' ) }
-							/>
-							<ToggleGroupControlOption
-								value="mobile"
-								label={ __( 'Mobile' ) }
-							/>
-							<ToggleGroupControlOption
-								value="always"
-								label={ __( 'Always' ) }
-							/>
-						</ToggleGroupControl>
+
+						{ ! hideOverlayControls && (
+							<>
+								<HStack className="wp-block-navigation__menu-inspector-controls__overlay-menu">
```

Comment:
Let's extract all these overlay controls to a dedicated component.

Reply:
or maybe a filter that filters the nav block edit and _adds_ the overlay controls if not in an overlay area?

Reply:
I'm curious - what would be the benefit of that approach? Filters are expensive and we have direct access to the block code.

Reply:
> Filters are expensive 

I don't think It has benefits other than making the block code a bit more lean IMO